AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -307,8 +307,7 @@ - (ASCellNode *)supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPat
       return nodesOfKindInSection[itemIndex];
     }
   }
-  ASDisplayNodeAssert(NO, @"Supplementary node should exist.  Kind = %@, indexPath = %@, collectionDataSource = %@", kind, indexPath, self.collectionDataSource);
-  return [[ASCellNode alloc] init];
+  return nil;
```

Comment:
Best to move the assertion to the place in the data source that requires a supplementary node — I can see changing this behaviour causing some problems.